src/hotspot/share/runtime/mutex.cpp

```diff
@@ -384,14 +384,21 @@ void Mutex::check_rank(Thread* thread) {
   if (owned_by_self()) {
     // wait() case
     Mutex* least = get_least_ranked_lock_besides_this(locks_owned);
-    // We enforce not holding locks of rank nosafepoint or lower while waiting.
+    // For JavaThreads, we enforce not holding locks of rank nosafepoint or lower while waiting
+    // because the held lock has a NoSafepointVerifier so waiting on a lower ranked lock will not be
+    // able to check for safepoints first with a TBIVM.
+    // For all threads, we enforce not waiting on the tty lock since that could block progress also.
```

Comment:
I think this should be s/on the tty lock/while holding the tty lock or below.

Reply:
fixed, thanks for the suggestion. It's more correct.